### PR TITLE
player/main: guard smtc.h include

### PR DESCRIPTION
**Pull request by llyyr**
Read this before you submit this pull request:
https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

Reading this link and following the rules will get your pull request reviewed
and merged faster. Nobody wants lazy pull requests.

### CODE REVIEW

File: player/main.c [hunk 1]
```diff
@@ -36,7 +36,10 @@
 #include "osdep/threads.h"
 #include "osdep/timer.h"
 #include "osdep/main-fn.h"
+
+#if defined(HAVE_WIN32_SMTC) && HAVE_WIN32_SMTC
```
Comment (sfan5) [inline player/main.c:40]: the `defined()` is not needed and also please move it next to the `HAVE_COCOA` thing

Comment (kasper93) [reply player/main.c]: > the defined() is not needed

It is needed. It is not checked on non-win32. Until https://github.com/mpv-player/mpv/pull/14624 from @Dudemanguy

Comment (llyyr) [reply player/main.c]: I just followed the pattern from line 405

Comment (sfan5) [reply player/main.c]: Not strictly. The C standard defines that undefined preprocessor macros are treated as `0`

Comment (Dudemanguy) [reply player/main.c]: We can cherrypick that the first commit from my PR that defines HAVE_WIN32_SMTC if you want.

Comment (llyyr) [reply player/main.c]: Seems non offensive to me, so I can add that if nobody minds

Comment (llyyr) [reply player/main.c]: Should I remove it and also remove defined() from the check later in the file then?

Comment (Dudemanguy) [reply player/main.c]: It avoids a warning if you care ¯\_(ツ)_/¯

**Comment by llyyr** [comment]
> ../player/main.c:405:5: error: "HAVE_WIN32_SMTC" is not defined, evaluates to 0 [-Werror=undef]

Should I use `defined()` here or just silence that error?

**Review by kasper93 — APPROVED** [review]

**Pull request merged**